IFMNowPlaying.m

```diff
@@ -23,11 +24,26 @@ @implementation IFMNowPlaying
 
 - (NSString *)_parseResponse:(NSData *)response
 {
-	NSString *nowPlayingString = [[NSString alloc] initWithData:response encoding:NSUTF8StringEncoding];
-	nowPlayingString = [nowPlayingString stringByReplacingOccurrencesOfString:@"&amp;" withString:@"&"];
-	NSArray *lines = [nowPlayingString componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]];
-	
-	return [lines objectAtIndexOrNil:2];
+
+    NSError *err = nil;
+    NSArray *jsonData = [NSJSONSerialization JSONObjectWithData:response options:NSJSONReadingAllowFragments error:&err];
+    
+    NSDictionary *dict = [jsonData objectAtIndex:0];
+    
+    NSString *sepa1 = @" - ";
+    NSString *sepa2 = @" | ";
+    NSNumber *yearInt = dict[@"year"];
+    NSLog(@"yearInt: %@",yearInt);
+    NSString *year = [NSString stringWithFormat:@"%@",yearInt];
+
+    NSString *nowPlaying1 = [dict[@"artist"] stringByAppendingString:[sepa1 stringByAppendingString:dict[@"track"]]];
```

Comment:
I see what you meant in your email. :) This whole thing could probably be replaced with one `+[NSString stringWithFormat:]`, although that will change the logic a bit. As opposed to Java, in Objective-C passing messages to `nil` is okay and just does nothing. So when you append to `dict[@"artist"]`, it can be `nil` (however unlikely), and the resulting `nowPlaying5` will be `nil` too.

Reply:
I basically put together code I found around, it could be the worst piece of code I ever developed :| 
Anyway, the potential issue here is for sure if one of the JSON attributes is null or empty.
In my mind I would have developed something like
`String nowPlaying = "";
for(String field : fields){
 if(isNonEmpty(field)){
nowPlaying += field;
}
}

boolean isNonEmpty(String stringToBeChecked){
 return stringToBechecked != null && stringToBeChecked.length > 0;
}`

and then the separation management

Reply:
Sure. If you can wait a few days, I'd be happy to work something out here. In the middle of xmas holidays with the family now. :)

Reply:
Of course! there is no rush, I am just working/playin on it since I am free now, but ready to revert all :)